scripts/Diffraction/isis_powder/routines/common.py

```diff
@@ -0,0 +1,165 @@
+    else:
+        load_monitor_ws = _load_sum_file_range(run_numbers_list=number_list, instrument=instrument)
+        load_monitor_ws = mantid.ExtractSingleSpectrum(InputWorkspace=load_monitor_ws, WorkspaceIndex=monitor_spectra)
+
+    return load_monitor_ws
+
+
+def subtract_sample_empty(ws_to_correct, empty_sample_ws_string, instrument):
+    output = ws_to_correct
+    if empty_sample_ws_string is not None:
+        empty_sample = load_current_normalised_ws(run_number_string=empty_sample_ws_string, instrument=instrument)
+        corrected_ws = mantid.Minus(LHSWorkspace=ws_to_correct, RHSWorkspace=empty_sample)
+        remove_intermediate_workspace(empty_sample)
+        output = corrected_ws
+    return output
+
+
+def _load_raw_files(run_number_string, instrument):
+    run_number_list = generate_run_numbers(run_number_string=run_number_string)
+    instrument.PEARL_setup_input_directories(run_number=run_number_list[0])
+    load_raw_ws = _load_sum_file_range(run_number_list, instrument)
+    return load_raw_ws
+
+
+def _load_sum_file_range(run_numbers_list, instrument):
+    read_ws_list = []
+
+    _check_load_range(list_of_runs_to_load=run_numbers_list)
+
+    for run_number in run_numbers_list:
+        file_name = instrument._generate_inst_file_name(run_number=run_number)
+        read_ws = mantid.Load(Filename=file_name, LoadLogFiles="0")
+        read_ws_list.append(mantid.RenameWorkspace(InputWorkspace=read_ws, OutputWorkspace=str(file_name)))
+
+    # Sum all workspaces
+    out_ws_name = "summed_range_load-" + str(g_ads_workaround["read_ws"])
+    g_ads_workaround["read_ws"] += 1
+    summed_ws = mantid.RenameWorkspace(InputWorkspace=read_ws_list[0], OutputWorkspace=out_ws_name)
+    for ws in read_ws_list[1:]:  # Skip first member
+        summed_ws = mantid.Plus(LHSWorkspace=summed_ws, RHSWorkspace=ws, OutputWorkspace=out_ws_name)
+        remove_intermediate_workspace(ws)
+
+    return summed_ws
+
+
+def _check_load_range(list_of_runs_to_load):
+    MAXIMUM_RANGE_LEN = 20  # If more than this number of runs is entered probably wrong
+    if len(list_of_runs_to_load) > MAXIMUM_RANGE_LEN:
+        raise ValueError("More than " + str(MAXIMUM_RANGE_LEN) + " runs were selected."
+                         " Found " + str(len(list_of_runs_to_load)) + " Aborting.")
+
+
+def generate_run_numbers(run_number_string):
+
+    # Check its not a single run
+    if isinstance(run_number_string, int) or run_number_string.isdigit():
+        return [int(run_number_string)]  # Cast into a list and return
+
+    # If its a string we must parse it
+    run_number_string = run_number_string.strip()
+    run_boundaries = run_number_string.replace('_', '-')  # Accept either _ or - delimiters
+    run_range_lists = __run_number_generator(processed_string=run_boundaries)
+    out_range = []
+    for range_list in run_range_lists:
+        out_range.extend(range_list)
+
+    return out_range
+
+
+def __run_number_generator(processed_string):
```

Comment:
Will it handle comma separated ranges e.g. `1-10,12,14-20` loads: `1,2,3,4....8,9,10,12,14,15,16....,18,19,20`? 
I'll add it to the refactoring [here](https://github.com/mantidproject/mantid/issues/18108) to experiment around with it